sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathProperties.cs

```diff
@@ -171,6 +171,41 @@ public class PathProperties
         /// </summary>
         public DateTimeOffset AccessTierChangedOn { get; internal set; }
 
+        /// <summary>
+        /// If this path represents a directory.
+        /// </summary>
+        public bool IsDirectory
+        {
+            get
+            {
+                Metadata.TryGetValue(Constants.DataLake.IsDirectoryKey, out string isDirectory);
+                if (isDirectory != null && isDirectory == bool.TrueString.ToLowerInvariant())
+                {
+                    return true;
+                }
+                return false;
+            }
+            internal set
+            {
+                // True
+                if (value)
+                {
+                    if (!Metadata.ContainsKey(Constants.DataLake.IsDirectoryKey))
+                    {
+                        Metadata.Add(Constants.DataLake.IsDirectoryKey, bool.TrueString.ToLowerInvariant());
+                    }
+                }
+                // False
+                else
+                {
+                    if (Metadata.ContainsKey(Constants.DataLake.IsDirectoryKey))
+                    {
+                        Metadata.Remove(Constants.DataLake.IsDirectoryKey);
```

Comment:
You don't need the check - Remove won't throw if not found.  https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.remove?view=netframework-4.8

Reply:
fixed.